src/ocrd/processor/base.py

```diff
@@ -323,24 +325,31 @@ def process_page_file(self, *input_files : OcrdFile) -> None:
         to handle cases like multiple fileGrps, non-PAGE input etc.)
         """
         log = getLogger('ocrd.processor.base')
-        input_pcgts : List[OcrdPage] = [None] * len(input_files)
+        input_pcgts : List[Optional[OcrdPage]] = [None] * len(input_files)
+        assert isinstance(input_files[0], (OcrdFile, ClientSideOcrdFile))
         page_id = input_files[0].pageId
         for i, input_file in enumerate(input_files):
+            assert isinstance(input_file, (OcrdFile, ClientSideOcrdFile))
             # FIXME: what about non-PAGE input like image or JSON ???
             log.debug("parsing file %s for page %s", input_file.ID, input_file.pageId)
             try:
-                input_pcgts[i] = page_from_file(input_file)
+                page_ = page_from_file(input_file)
+                assert isinstance(page_, PcGtsType)
+                input_pcgts[i] = page_
             except ValueError as e:
```

Comment:
```suggestion
            except (AssertionError, ValueError) as e:
```

Reply:
Can this ever happen, ie. can `page_from_file(with_etree=False)` ever return anything other than a `PcGtsType`? I think if that was ever the case, we'd want that `AssertionError` to be raised because then we'd have broken something.

Reply:
You're right – it cannot happen. But then what is the assertion good for – satisfying the type checker?

Reply:
First, my curiosity that I understand the behavior correctly. But secondly, yes, the type checker  ;)

Reply:
But reading this again, I should have used `OcrdPage` not `PcGtsType`, which is just an alias but we use `OcrdPage` in the method typing.

Reply:
Yes. Feel free to change in #1240.